scripts/west_commands/runners/esp32.py

```diff
@@ -84,16 +85,24 @@ def do_create(cls, cfg, args):
         return Esp32BinaryRunner(
             cfg, args.esp_device, boot_address=args.esp_boot_address,
             part_table_address=args.esp_partition_table_address,
-            app_address=args.esp_app_address, baud=args.esp_baud_rate,
-            flash_size=args.esp_flash_size, flash_freq=args.esp_flash_freq,
-            flash_mode=args.esp_flash_mode, espidf=espidf,
-            bootloader_bin=args.esp_flash_bootloader,
+            app_address=args.esp_app_address, erase=args.erase,
+            baud=args.esp_baud_rate, flash_size=args.esp_flash_size,
+            flash_freq=args.esp_flash_freq, flash_mode=args.esp_flash_mode,
+            espidf=espidf, bootloader_bin=args.esp_flash_bootloader,
             partition_table_bin=args.esp_flash_partition_table)
 
     def do_run(self, command, **kwargs):
         self.require(self.espidf)
 
-        cmd_flash = [self.espidf, '--chip', 'auto']
+        # Add Python interpreter
+        cmd_flash = [sys.executable]
+
+        if self.erase is True:
+            cmd_flash.extend([self.espidf, 'erase_flash'])
+            self.check_call(cmd_flash)
+            return
```

Comment:
Expected behavior when the user runs `west flash --erase`:

- erase the flash on the target device, then
- flash the binary in the build directory

Please adjust the PR to match the expected behavior. Currently it is only doing the first step; it needs to do both.

Reply:
Thanks!